tests/gamma/csm_observability_test.py

```diff
@@ -45,39 +100,389 @@ def setUpClass(cls):
+        with self.subTest("6_check_metrics_label_keys"):
+            for metric in _HISTOGRAM_CLIENT_METRICS:
+                metric_labels = all_results[metric][0].metric.labels
+                for label_key in [
+                    "csm_mesh_id",
+                    "csm_remote_workload_canonical_service",
+                    "csm_remote_workload_cluster_name",
+                    "csm_remote_workload_location",
+                    "csm_remote_workload_name",
+                    "csm_remote_workload_namespace_name",
+                    "csm_remote_workload_project_id",
+                    "csm_remote_workload_type",
+                    "csm_service_name",
+                    "csm_service_namespace_name",
+                    "csm_workload_canonical_service",
+                    "grpc_method",
+                    "grpc_status",
+                    "grpc_target",
+                    "otel_scope_name",
+                    "otel_scope_version",
+                    "pod",
+                ]:
+                    self.assertIn(label_key, metric_labels)
+
+            for metric in _HISTOGRAM_SERVER_METRICS:
+                metric_labels = all_results[metric][0].metric.labels
+                for label_key in [
+                    "csm_mesh_id",
+                    "csm_remote_workload_canonical_service",
+                    "csm_remote_workload_cluster_name",
+                    "csm_remote_workload_location",
+                    "csm_remote_workload_name",
+                    "csm_remote_workload_namespace_name",
+                    "csm_remote_workload_project_id",
+                    "csm_remote_workload_type",
+                    "csm_workload_canonical_service",
+                    "grpc_method",
+                    "grpc_status",
+                    "otel_scope_name",
+                    "otel_scope_version",
+                    "pod",
+                ]:
+                    self.assertIn(label_key, metric_labels)
+
+            for metric in _COUNTER_CLIENT_METRICS:
+                metric_labels = all_results[metric][0].metric.labels
+                for label_key in [
+                    "grpc_method",
+                    "grpc_target",
+                    "otel_scope_name",
+                    "otel_scope_version",
+                    "pod",
+                ]:
+                    self.assertIn(label_key, metric_labels)
+
+            for metric in _COUNTER_SERVER_METRICS:
+                metric_labels = all_results[metric][0].metric.labels
+                for label_key in [
+                    "grpc_method",
+                    "otel_scope_name",
+                    "otel_scope_version",
+                    "pod",
+                ]:
+                    self.assertIn(label_key, metric_labels)
+
+        # Testing whether each metric has the right set of monitored resource
+        # label keys
+        with self.subTest("7_check_resource_label_keys"):
```

Comment:
I don't care too much about this either, but I'm fine with leaving it in.

Reply:
I will leave it as is coz it helps verify the GMP part of things.